lib/nixos-system.nix

```diff
@@ -1,15 +1,14 @@
 { self, lib, home-manager }:
 
-{ modules ? [ ], specialArgs ? { }, ... } @ args:
-lib.nixosSystem (args // {
-  specialArgs = lib.mergeAttrs
+{ modules ? [ ], modulesArgs ? { }, ... } @ args:
+lib.nixosSystem (lib.recursiveUpdate (removeAttrs args [ "modulesArgs" ]) {
+  specialArgs = { mode = "NixOS"; } // modulesArgs;
```

Comment:
I think the `mode` can go into the _module.args but I agree to not bundle it in this PR.

Reply:
Actually I tried this, as I moved `home-manager` to `_module.args`, and it gives me an infinite recursion. Probably because the evaluation of `config` depends on conditionals using it.